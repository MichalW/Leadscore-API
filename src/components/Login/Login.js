import React from 'react';
import {
  Button, Container, Form, Grid, Input, Label,
} from 'semantic-ui-react';

const Login = () => (
  <Container>
    <Grid columns={3} centered>
      <Grid.Row>
        <Grid.Column>
          <Form>
            <Form.Field>
              <Label>Login</Label>
              <Input placeholder="Login" />
            </Form.Field>

            <Form.Field>
              <Label>Password</Label>
              <input placeholder="Password" type="password" />
            </Form.Field>

            <Button type="submit">Login</Button>
          </Form>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

export default Login;
