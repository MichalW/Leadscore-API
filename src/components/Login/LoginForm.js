import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = (event) => {
    this.setState({username: event.target.value});
  };

  onChangePassword = (event) => {
    this.setState({password: event.target.value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          <Button type="submit" fluid primary>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
