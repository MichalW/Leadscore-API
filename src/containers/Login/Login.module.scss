@import "../../styles/colors";

.wrapper {
  background-color: $color-blue;
  min-height: 100vh;
}
