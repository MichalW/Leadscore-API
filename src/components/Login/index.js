import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Login from './Login';
import {signInRequest} from '../../actions/authentication';

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(null, mapDispatchToProps)(Login);
