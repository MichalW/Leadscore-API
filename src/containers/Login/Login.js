import React from 'react';
import {Grid, Header} from 'semantic-ui-react';

import LoginForm from '../../components/LoginForm';
import styles from './Login.module.scss';

const Login = () => (
  <Grid
    centered
    className={styles.wrapper}
    columns={3}
    style={{minHeight: '100vh'}}
    verticalAlign="middle"
  >
    <Grid.Row>
      <Grid.Column>
        <Header as="h1">
          <strong>Lead</strong>
          score
        </Header>

        <LoginForm />
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

export default Login;
