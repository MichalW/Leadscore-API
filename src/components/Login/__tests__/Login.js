import React from 'react';
import renderer from 'react-test-renderer';

import Login from '../Login';

it('renders Login without crashing', () => {
  const tree = renderer
    .create(<Login signInRequest={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
