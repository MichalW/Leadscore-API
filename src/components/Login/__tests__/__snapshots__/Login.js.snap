// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login without crashing 1`] = `
<div
  className="ui centered middle aligned three column grid"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="ui segment"
        >
          <div
            className="field"
          >
            <div
              className="ui label"
              onClick={[Function]}
            >
              Login
            </div>
            <div
              className="ui input"
            >
              <input
                onChange={[Function]}
                placeholder="Login"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui label"
              onClick={[Function]}
            >
              Password
            </div>
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            className="ui fluid button"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
